### ISSUE TITLE
Clarification on Commercial License Impact of LayoutLMv3ImageProcessor within UdopProcessor

### ISSUE THREAD

**Issue by Arjunexperion** [NielsRogge/Transformers-Tutorials#484]
Hi team,

I have a question regarding licensing and commercial usage.

Since UdopProcessor internally uses LayoutLMv3ImageProcessor (as part of resizing, rescaling, normalizing document images, and applying OCR), and given that LayoutLMv3 itself is not licensed for commercial use, I would like to clarify:

➡️ If I use UdopProcessor for fine-tuning UDOP and plan to deploy it in a commercial setting, will the dependency on LayoutLMv3ImageProcessor affect the commercial viability of using UDOP?

In other words, does the inclusion of LayoutLMv3ImageProcessor within UdopProcessor impose any commercial licensing restrictions on the UDOP model?

Thank you in advance!

**Comment by NielsRogge** [comment]
That's a good question. It might be that only the weights are subject to the non-commercial license, and the code can be used, but I would recommend discussing that with a legal manager.

**Comment by Arjunexperion** [comment]
Thankyou for your response. And also can we implement a custom image processor here for Udop instead of relying on layoutlmv3imageprocessor . If yes How can we do it? And also can you recommend some models for finetuning for document classification and text extraction which can be used for commercial purpose.
